### ISSUE TITLE
unable to review text in text fields with a screen reader

### ISSUE THREAD

**Issue by crentava** [authpass/authpass#216]
now that the unlabeled button problem has been fixed from what I’ve heard I’d like to try to solve one of the final accessibility problems. The text fields in your application do not allow us to review any input with a screen reader. are you using the default flutter text field or a custom one? If you’re using a custom one could you please edit it’s semantics properties? and what I mean by review text is, in a normal text field, a screen reader will be able to navigate up and down by character or word to read
thank you

**Comment by hpoul** [comment]
Well, actually they are default flutter text fields. I'm not completely sure how to change that.. happens this on all text fields in the app (e.g. also in the 'search' of the password list) or only in the input fields for passwords?

**Comment by crentava** [comment]
it happens in all text fields. The reason I asked about the text field is because I saw other apps made in flutter who’s text fields are perfectly fine. But yours along with a couple others didn’t seem to work so I thought you guys were using custom text field widgets

**Comment by hpoul** [comment]
how can I try this out? Is this the "Spoken Content" feature on iOS?

**Comment by crentava** [comment]
no. It’s a feature called voiceover. It’s in accessibility settings. But be warned, the way you navigate will be different

**Comment by hpoul** [comment]
ah okay, just discovered that this isn't available in the ios simulator, that's why i didn't find it.. have to check it on a physical device

**Comment by crentava** [comment]
this may help
[text field semantics properties](https://api.flutter.dev/flutter/semantics/SemanticsProperties/textField.html)

**Comment by hpoul** [comment]
ok i think i understand what you are referring to.. usually in a text field when swiping left/right it would go to previous/next character.. but in authpass it would go to the next/previous button, instead of changing the cursor.. right?

**Comment by crentava** [comment]
something like that. The curser would be moved by swiping up and down to read by character or if you change the navigation settings, by word or line. I think that somehow the semantics for these textfields are messed up but you got the gist

**Comment by hpoul** [comment]
funny.. there was a related flutter issue which was recently closed: https://github.com/flutter/engine/pull/23776 it was merged into flutter a day too late for the current dev channel .. i'll try the master channel if this fixes things

**Comment by crentava** [comment]
will that only work for voiceover? Or will it also work for android users as well?

**Comment by hpoul** [comment]
is there a problem on android? haven't tried it

**Comment by crentava** [comment]
same problem as on iOS I think. I’ll have to test it though so don’t quote me on that

**Comment by hpoul** [comment]
i've just updated the testflight version with the latest flutter master version. looks good to me

**Comment by crentava** [comment]
thank you, That fixed it. And one final question, when do the dev builds get released to the App Store?

**Comment by hpoul** [comment]
once I release them ;-) but i'm not sure i want to release a version which uses a flutter master channel version.. and there is usually only a new dev release every 2-3 weeks 🤔️

**Comment by crentava** [comment]
so… Will this version that you release to the app stores roll back the changes to the keyboard because it’s not on the master channel?

**Comment by hpoul** [comment]
i think i will wait another week to see if there is another flutter dev release.

**Comment by hpoul** [comment]
@crentava fyi, i've just published a new version to testflight which uses the latest flutter dev channel release. maybe you could give it another try if everything's still working? 😅️

**Comment by crentava** [comment]
@hpoul everything is working perfectly👍

**Comment by hpoul** [comment]
It's now included in 1.7.11 which i've just released to the app store and play store. If you have any further problems don't hesitate to open a new issue. thanks!
